### PR TITLE
[Workflow] Added new validator to make sure each place has unique translation names

### PR DESCRIPTION
**Pull request by Nyholm**
| Q             | A
| ------------- | ---
| Branch?       | 3.2
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | 
| License       | MIT
| Doc PR        | n/a

A definition where a place that has transitions with the same name is an invalid definition. 

```yaml
invalid1: 
  places: ['a', 'b', 'c']
  transitions: 
    t1: 
      from: a
      to: b
    t1: 
      from: a
      to: c

valid1: 
  places: ['a', 'b', 'c']
  transitions: 
    t1: 
      from: a
      to: b
    t2: 
      from: a
      to: c

valid2: 
  places: ['a', 'b', 'c']
  transitions: 
    t1: 
      from: a
      to: b
    t1: 
      from: b
      to: c

valid3: 
  places: ['a', 'b', 'c', 'd']
  transitions: 
    t1: 
      from: ['a', 'b']
      to: d
    t2: 
      from: ['a', 'b']
      to: c
```

FYI @lyrixx

### CODE REVIEW

**Review by lyrixx — APPROVED** [review]
:+1:

**Comment by lyrixx** [comment]
**Note**: It should be merge in 3.2 and not master.

File: src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Compiler/ValidateWorkflowsPass.php [hunk 1]
```diff
@@ -16,6 +16,7 @@
 use Symfony\Component\DependencyInjection\Exception\RuntimeException;
 use Symfony\Component\Workflow\Validator\DefinitionValidatorInterface;
 use Symfony\Component\Workflow\Validator\StateMachineValidator;
+use Symfony\Component\Workflow\Validator\UniqueTransitionNameValidator;
```
Comment (nicolas-grekas) [inline src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Compiler/ValidateWorkflowsPass.php]: missing composer.json update I guess, for ^3.2.3

Comment (Nyholm) [inline src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Compiler/ValidateWorkflowsPass.php]: There is no reference to symfony/workflow at all in the composer.json of the framework bundle. Should I still add one?

**Comment by ro0NL** [comment]
What about moving it to `WorkflowValidator`? The `StateMachineValidator` already enforces this right?

**Comment by Nyholm** [comment]
Thank you @ro0NL. You are correct. The StateMachineValidator has this test already but it is missing in the WorkflowValidator. 

I will merge this test to the Workflow validator.

**Comment by fabpot** [comment]
Looks like a new feature to me, not a bug fix. I think it should be merged in 3.3 instead.

**Comment by Nyholm** [comment]
> Looks like a new feature to me, not a bug fix.

This PR makes sure the Definition is not invalid. That is why I marked it as a "bugfix". But one could argue that all PRs to the Validator as bugfixes... Anyhow. 

Since this check exist for the [StateMachineValidator](https://github.com/symfony/symfony/blob/v3.2.2/src/Symfony/Component/Workflow/Validator/StateMachineValidator.php#L37-L43), I've updated the PR to include this check in the Workflow validator only.

**Comment by nicolas-grekas** [comment]
Generally : any new visible interface => feature.
Now that this just tweaks an edge case behavior only, it may much likely qualify as bug fix :)

**Comment by lyrixx** [comment]
For me it's a bug fix too, because if the definition is not valid, the workflow will not work correctly.

**Comment by fabpot** [comment]
Thank you @Nyholm.

**Pull request closed without merging**